crates/uv/tests/pip_compile.rs

```diff
@@ -9272,6 +9272,229 @@ fn resolve_configuration() -> Result<()> {
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --exclude-newer 2024-03-25T00:00:00Z requirements.in
+    anyio==3.0.1
+        # via -r requirements.in
+    idna==3.6
+        # via anyio
+    sniffio==1.3.1
+        # via anyio
+
+    ----- stderr -----
+    Resolved 3 packages in [TIME]
+    "###
+    );
+
+    // Add a local configuration to generate hashes.
+    let config = context.temp_dir.child("uv.toml");
+    config.write_str(indoc::indoc! {r"
+        [pip]
+        generate-hashes = true
+    "})?;
+
+    // Resolution should use the lowest direct version and generate hashes.
+    uv_snapshot!(context.compile()
+        .arg("requirements.in")
+        .env("XDG_CONFIG_HOME", xdg.path()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --exclude-newer 2024-03-25T00:00:00Z requirements.in
+    anyio==3.0.1 \
+        --hash=sha256:1ef7622396ab55829d4236a6f75e2199df6d26a4ba79bea0cb942a5fd2f79a23 \
+        --hash=sha256:ed71f7542ef39875b65def219794d9dcb0a48c571317b13612c12b1f292701b5
+        # via -r requirements.in
+    idna==3.6 \
+        --hash=sha256:9ecdbbd083b06798ae1e86adcbfe8ab1479cf864e4ee30fe4e46a003d12491ca \
+        --hash=sha256:c05567e9c24a6b9faaa835c4821bad0590fbb9d5779e7caa6e1cc4978e7eb24f
+        # via anyio
+    sniffio==1.3.1 \
+        --hash=sha256:2f6da418d1f1e0fddd844478f41680e794e6051915791a034ff65e5f100525a2 \
+        --hash=sha256:f4324edc670a0f49750a81b895f35c3adb843cca46f0530f79fc1babb23789dc
+        # via anyio
+
+    ----- stderr -----
+    Resolved 3 packages in [TIME]
+    "###
+    );
+
+    // Add a local configuration to override the user configuration.
+    let config = context.temp_dir.child("uv.toml");
+    config.write_str(indoc::indoc! {r#"
+        [pip]
+        resolution = "highest"
+    "#})?;
+
+    // Resolution should use the highest version.
+    uv_snapshot!(context.compile()
+        .arg("requirements.in")
+        .env("XDG_CONFIG_HOME", xdg.path()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --exclude-newer 2024-03-25T00:00:00Z requirements.in
+    anyio==4.3.0
+        # via -r requirements.in
+    idna==3.6
+        # via anyio
+    sniffio==1.3.1
+        # via anyio
+
+    ----- stderr -----
+    Resolved 3 packages in [TIME]
+    "###
+    );
+
+    // However, the user-level `tool.uv.pip` settings override the project-level `tool.uv` settings.
+    // This is awkward, but we merge the user configuration into the workspace configuration, so
+    // the resulting configuration has both `tool.uv.pip.resolution` (from the user configuration)
+    // and `tool.uv.resolution` (from the workspace settings), so we choose the former.
```

Comment:
Not really ideal, I think.

Reply:
It's a little awkward but at least it's consistent, maybe we should warn in that case? Are we going to have more complex merging rules for the pip namespace from the user-level config?